### DISCUSSION TITLE
Trouble with grain boundaries

### DISCUSSION THREAD

**Discussion by Emil-cmd** [mtex-toolbox/mtex#1566] [Ask Anything]
Hello. I am having two issues that concern grain boundaries. 

The first one: 
I have a dataset showing to large deformed grains with a boundary in between. For some reason, even though the HAGB limit is set to 10 degrees, for this specific data mtex still set this not as a boundary but an innerBoundary. Line scans in both CHANNEL 5 and mtex shows that the misorientation is more than 40 degrees. The boundary is set as a HAGB with the same conditions in CHANNEL 5. 
In the included IPF figure, HAGB are in black and LAGB are in gray. 
Can supply the data if necessary or beneficial for trouble shooting. 
![IPF](https://user-images.githubusercontent.com/54838740/191492220-2f92fc98-dbef-4fae-bd06-41625a5ed503.png)
![Linescan](https://user-images.githubusercontent.com/54838740/191492245-afd94a3f-6d64-4a0c-84ac-4569e5888588.png)

The other issue I have is merging Sigma3 twins. I follow the instructions on the webpage and it works fine if doing it over the complete dataset. However I am only interested in doing this for a subset of grains. However for some samples I get the following error 
**Error using sparse
Index exceeds array bounds.

Error in grain2d/merge (line 68)
    A = A | sparse(mergeId(:,1),mergeId(:,2),1,maxId+1,maxId+1);**

This error occurs if I first extract the EBSD data to only include the interesting grains 
ebsd_subset = ebsd_full(grains_subset); 
CSL =  grains_subset(angle(grains_subset.boundary.misorientation,CSL(3,ebsd_subset.CS)) < 3*degree); 
[mergedGrains,parentIds] = merge(grains_subset,CSL);

or if I take the complete EBSD data and only try and merge the subset of grains 
ebsd = ebsd;
CSL =  grains(angle(grains.boundary.misorientation,CSL(3,ebsd.CS)) < 3*degree); 
[mergedGrains,parentIds] = merge(grains_subset,CSL);

Can anyone help me understand why I get these two issues? 

Thank you in advance

**Comment by kilir** [comment]
Hi,

inner boundaries can have misorientation angles larger larger than the given threshold. They just won't become grain boundaries since they do not inscribe a closed shape.

Cheers,
Rüdiger

**Comment by Emil-cmd** [comment]
Hello Rüdiger
I see, but just to make sure. The issue in my case (if you can see the attached figure to my original post) is that the boundary in question does not inscribe a closed shape since the boundary goes across the entire map?

Med vänlig hälsning
Emil Eriksson, Doktorand
Institutionen för Fysik
Avdelningen för Mikrostrukturfysik

9 okt. 2022 kl. 17:10 skrev Rüdiger Kilian ***@***.***>:

﻿

Hi,

inner boundaries can have misorientation angles larger larger than the given threshold. They just won't become grain boundaries since they do not inscribe a closed shape.

Cheers,
Rüdiger

**Reply by zmichels** [reply]
There seems to be some conflation between LAGB and InnerGB. They are not the same. LAGBs are defined by the magnitude of the misorientation angle... and inner boundaries are boundaries that do not inscribe a complete area in the map to define a grain. Two different things. Grain boundaries will always be greater than the threshold set because that's how you inform the grain boundary construction. Inner boundaries can have any angle. Perhaps in many cases they will be LAGB, but they don't have to be based on the definition of an innderBoundary. You can still locate HAB that are inner boundaries (as you have).
